### DISCUSSION TITLE
How to be indexed in Google Search Console

### DISCUSSION THREAD

**Discussion by titaliu1224** [cotes2020/jekyll-theme-chirpy#1176] [Q&A]
**NOTE:** Before you start, the following should be completed.

- Read [Wiki][wiki] to understand the usage and the correct effect of functional design.
- Make sure no [similar issue(including closed ones)][issues] exists.
- Try to find the answer on [Jekyll Forum][forum] and [StackOverflow][stack_overflow].

[wiki]: https://github.com/cotes2020/jekyll-theme-chirpy/wiki
[issues]: https://github.com/cotes2020/jekyll-theme-chirpy/issues?q=is%3Aissue
[forum]: https://talk.jekyllrb.com/
[stack_overflow]: https://stackoverflow.com/questions/tagged/jekyll

## Description

I submitted a sitemap to Google Search Console a few months ago. The sitemap shows that Google has crawled 85 pages, but only 3 pages appear in the "Pages" report. I have also used the URL inspection tool to request indexing, but the pages are still not indexed. I have contacted Google Search Console support, but I have not received a response.
Is there anything else I can do besides submitting the sitemap and waiting? If I need to create a robot.txt file, can I just add a file in the root directory?

我在幾個月前提交了site map 到 Google Search Console。site map 顯示 Google 已經爬取了 85 個頁面，但在「頁面」報表中只顯示了 3 個頁面。也有使用 URL 檢查工具來要求建立索引，但頁面仍未建立索引。即便聯絡 Google Search Console 支援，也還沒有收到回覆，請問除了把 sitemap 提交上去並等待，我是否還要做些什麼？如果需要建立 robot.txt 的話，直接在根目錄新增一個檔案就好了嗎？

> [my code](https://github.com/titaliu1224/titaliu1224.github.io)

## Operations you have already tried

I have also used the URL inspection tool to request indexing, but the pages are still not indexed. Only the homepage and ABOUT pages are indexed...

手動提交網址審查也沒有頁面被索引進去，只有一開始的主頁和 ABOUT......

## Logs/Screenshots

![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/71343865/ef8a8382-fb1a-478f-93a2-ae2319ab0a33)
![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/71343865/a94188e2-c65a-4a19-aced-e31c26899459)
![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/71343865/d2791187-4e1d-4090-97f8-4ae9ef50f0f4)

**Comment by cotes2020** [comment]
You don't need to create a `robots.txt` manually, since Chirpy has already generated it for you. See: `https://<your-site-domain>/robots.txt`.

The only noticeable issue so far is that your `sitemap.xml` URLs has a lot of `//` in it. I'm not sure if this is affecting the way Google crawls your site, but you should fix this by removing the `/` from the tail of `url` in your Jekyll config.

![Screen Shot 2023-08-14 at 19 38 39](https://github.com/cotes2020/jekyll-theme-chirpy/assets/11371340/be4aa290-199f-4c73-9849-d0855fbf4bdf)


Being "indexed" is the sign of being crawled, therefore you only have 3 pages crawled so far. Contacting Google is one of the right ways to handle it, alternatively you can click on the :grey_question: or  :grey_exclamation: button on the Search Console to get Google's tips and guidelines.

**Reply by titaliu1224** [reply]
I have confused about why there is `//` not `/` for a long time. Thank you for the imformation!

**Reply by patrykpalej** [reply]
Is it possible to mark some directories/pages not to be indexed?

**Reply by NichtsHsu** [reply]
Add `Disallow: /xxxx` to `assets/robots.txt`.

**Reply by nyhryan** [reply]
How long does it take for posts to get indexed? I've been waiting ages to get the posts indexed. I want to make sure posts get indexed eventually. I want to just write posts as much I want, not having to worry about my posts not being exposed to the internet.

**Comment by NichtsHsu** [comment]
GSC 的机制一直很迷惑，我sitemap提交一年了一直都是 couldn't fetch。我以前还在 [community 发帖问过](https://support.google.com/webmasters/thread/200080148?hl=en&sjid=7063822248708061526-AP)，得到的回复说 Google 不会为所有的页面建立索引，我就摆烂了。

**Reply by titaliu1224** [reply]
我的這個問題他們就寫得很模糊，說可能是網站流量、網站品質等等的問題，還說也有可能 Google 就不索引了，就很奇怪

**Comment by kungfux** [comment]
I invite you to consider my [thoughts](https://github.com/cotes2020/jekyll-theme-chirpy/discussions/1514#discussioncomment-8635995) on this subject in another thread.
